### PR TITLE
Local Storage Backend

### PR DESCRIPTION
**Pull request by Sipondo**
#262

### CODE REVIEW

File: modyn/selector/internal/storage_backend/local/local_storage_backend.py
Comment (MaxiBoether) [inline modyn/selector/internal/storage_backend/local/local_storage_backend.py:6]: Is there a reason we have the LocalStorageBackend in common and then import it here? I am considering either having the CPP wrapper in the storage component or alternatively, both within the same class here. I think it's not a common utility but rather specific for the storage (unlike the triggersamplestorage, where this historaically probably stems from). I'd be happy if you could adjust it in one of those two ways (and move the tests appropriately)

File: modyn/selector/internal/storage_backend/local/local_storage_backend.py
Comment (MaxiBoether) [inline modyn/selector/internal/storage_backend/local/local_storage_backend.py:11]: should we make this a modyn system parameter?

File: modyn/tests/selector/internal/storage_backend/local/test_local_storage_backend.py
Comment (MaxiBoether) [inline modyn/tests/selector/internal/storage_backend/local/test_local_storage_backend.py:1]: Thanks for the tests here - we need to slightly adjust this depending on how we decide to change the structure: If we have the CPP Python wrapper still in a separate file, let's test that individually and then the LocalStorageBackend, which uses this wrapper, separately.

Additionally, similarily to the triggersamplestorage, it would be great to have a couple of C++ tests as well to have the sanitizer check this as well.

**Review by MaxiBoether — APPROVED** [review]
Thanks Ties! I am pre-approving this but have two comments:

1) the structure of the CPP wrapper (it should go into the storage component instead of common I think, since it's not a common utility, unlike the TSS). 

2) a couple of C++ tests similar to the TSS would be amazing

Thank you so much. After adjusting this feel free to merge.

File: modyn/common/local_storage_backend/__init__.py
Comment (Sipondo) [reply modyn/common/local_storage_backend/__init__.py:2]: Done

File: modyn/selector/internal/storage_backend/local/local_storage_backend.py
Comment (Sipondo) [reply modyn/selector/internal/storage_backend/local/local_storage_backend.py:11]: yeah we should - where should I add it?

Comment (Sipondo) [reply modyn/selector/internal/storage_backend/local/local_storage_backend.py:14]: done

File: modyn/selector/internal/storage_backend/local/local_storage_backend.py
Comment (Sipondo) [reply modyn/selector/internal/storage_backend/local/local_storage_backend.py]: done

File: modyn/tests/selector/internal/storage_backend/local/test_local_storage_backend.py
Comment (Sipondo) [reply modyn/tests/selector/internal/storage_backend/local/test_local_storage_backend.py:1]: Done

**Review by MaxiBoether — APPROVED** [review]
Thank you! Only the Dockerfile thing I think we need to change, everything else LGTM! Thanks!

File: docker/Storage/Dockerfile
Comment (Sipondo) [reply docker/Storage/Dockerfile:16]: done

**Pull request merged**
